### PR TITLE
AYS-582 | Restrict Non-Active Users from Sending Password Reset Requests

### PR DESCRIPTION
**Pull request by egehanasal**
### Checklist

Before submitting your pull request, ensure the following:

- [X] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#branch-naming-conventions).

- [X] **Local Testing:**
    - I have tested my changes locally on Postman, and they are working as expected.

- [X] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [X] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes made in Postman.

- [X] **Testing:**
    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [X] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [X] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

### CODE REVIEW

**Review by filizhelvaci — APPROVED** [review]

**Review by filizhelvaci — APPROVED** [review]

**Review by filizhelvaci — APPROVED** [review]

**Review by filizhelvaci — APPROVED** [review]

File: src/main/java/org/ays/auth/exception/AysUserNotActiveByStatusException.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.ays.auth.model.enums.AysUserStatus;
+import org.ays.common.exception.AysAuthException;
+
+import java.io.Serial;
+
+/**
+ * Exception thrown when an operation is attempted on a user who is not active due to their current status.
+ */
+public final class AysUserNotActiveByStatusException extends AysAuthException {
```
[the remark this answers is not in the document]
Comment (egehanasal) [reply src/main/java/org/ays/auth/exception/AysUserNotActiveByStatusException.java:11]: cok mantikli, tamamdir oyle yapayim, normalde by status ve by id gibi exceptionlar olusturmusuz onu devam ettirmek istemistim

[the remark this answers is not in the document]
Comment (agitrubard) [reply src/main/java/org/ays/auth/exception/AysUserNotActiveByStatusException.java:11]: @egehanasal oradaki konu şu, mesela ID ile arama yapıp kullanıcıyı bulamadığımızda ID ile arattık bulamadık demek için By ID diye yazmıştık. Burada böyle bir durum tam olarak yok

**Review by alpemreelmas — APPROVED** [review]

File: src/main/java/org/ays/auth/service/impl/AysUserPasswordServiceImpl.java [hunk 2]
```diff
@@ -115,6 +117,10 @@ public void createPassword(final String passwordId,
         final AysUser user = userReadPort.findByPasswordId(passwordId)
                 .orElseThrow(() -> new AysUserPasswordDoesNotExistException(passwordId));
 
+        if(!user.isActive()) {
+            throw new AysUserNotActiveByStatusException(user.getStatus());
+        }
```
[the remark this answers is not in the document]
Comment (egehanasal) [reply src/main/java/org/ays/auth/service/impl/AysUserPasswordServiceImpl.java]: degisiklikleri yaptim 🙌

File: src/test/java/org/ays/auth/service/impl/AysUserPasswordServiceImplTest.java [hunk 3]
```diff
@@ -185,6 +187,40 @@ void givenValidForgotPasswordRequest_whenEmailDoesNotExist_thenThrowAysEmailAddr
...
+
+        AysUser mockUser = new AysUserBuilder()
+                .withValidValues()
+                .withEmailAddress(mockForgotPasswordRequest.getEmailAddress())
+                .withStatus(AysUserStatus.PASSIVE)
+                .build();
+
+        // When
+        Mockito.when(userReadPort.findByEmailAddress(Mockito.anyString()))
+                .thenReturn(Optional.of(mockUser));
```
Comment (agitrubard) [inline src/test/java/org/ays/auth/service/impl/AysUserPasswordServiceImplTest.java:205]:
```suggestion
        // When
        AysUser mockUser = new AysUserBuilder()
                .withValidValues()
                .withEmailAddress(mockForgotPasswordRequest.getEmailAddress())
                .withStatus(AysUserStatus.PASSIVE)
                .build();
                
        Mockito.when(userReadPort.findByEmailAddress(Mockito.anyString()))
                .thenReturn(Optional.of(mockUser));
```

**Review by agitrubard — CHANGES_REQUESTED** [review]

File: src/test/java/org/ays/auth/service/impl/AysUserPasswordServiceImplTest.java (hunk 3 above)
Comment (egehanasal) [reply src/test/java/org/ays/auth/service/impl/AysUserPasswordServiceImplTest.java:205]: done

**Review by MenekseYuncu — APPROVED** [review]

**Pull request merged**